File: parciales/ED1-Parcial1-2021-1--001/codigo/Permutaciones.java
import java.util.ArrayList;
class Permutaciones{
public static ArrayList<String> solve(String s) {
  ArrayList<String> sol = new ArrayList();
  tmp("", s, 0, s.length() - 1, sol);
  return sol;
}
public static void tmp(String l,String s,int i,int j, ArrayList<String> sol)
{
  if (i > j) {
    sol.add(l);
    return;
  }
  String c = "" + s.charAt(i);
  c = c.toLowerCase();
  tmp(l + c, s, i + 1, j, sol);
  String k = "" + s.charAt(i);
  k = k.toUpperCase();
  if (!c.equals(k)) {
    tmp(l + k, s, i + 1, j, sol);
  }
}
public static void main(String[] args){
  System.out.println(solve("as22b"));
  System.out.println(solve("123"));
  System.out.println(solve("1B4"));
}
}

File: parciales/ED1-Parcial1-2021-1--001/codigo/Permutaciones.py
def solve(s):
  sol = [];
  tmp("", s, 0, len(s) - 1, sol)
  return sol

def tmp( l, s, i, j, sol):

  if i > j:
    sol.append(l)#..........................
    return

  c = "" + s[i]
  c = c.lower()
  tmp(l + c, s, i + 1, j, sol)
  k = "" + s[i]
  k = k.upper()
  if not c == k:
    tmp(l + k, s, i + 1, j, sol)#...........................

def __main__():
  print(solve("as22b"));
  print(solve("123"));
  print(solve("1B4"));

__main__()

